src/main/scala/io/scalac/tezos/translator/routes/LoginRoute.scala

```diff
@@ -0,0 +1,23 @@
+package io.scalac.tezos.translator.routes
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.{HttpResponse, StatusCodes}
+import akka.http.scaladsl.server.{Directive, Route}
+import io.scalac.tezos.translator.model.UserCredentialsDTO
+import io.scalac.tezos.translator.routes.util.DTOValidationDirective.withDTOValidation
+import io.scalac.tezos.translator.service.UserService
+
+import scala.util.Success
+
+class LoginRoute(userService: UserService)(implicit as: ActorSystem) extends HttpRoutes with JsonHelper {
+
+  override def routes: Route =
+    (pathPrefix("login") & pathEndOrSingleSlash & validateCredentialsFormat & post) { credentials =>
+      onComplete(userService.authenticateAndCreateToken(credentials.username, credentials.password)) {
+        case Success(Some(token)) => complete(token)
+        case _ => complete(HttpResponse(status = StatusCodes.Forbidden))
```

Comment:
Right now the information we send to a user is either "here you have a token" or "this is forbidden". Where in one case there f.ex. are network failures and incorrect password
What would you say to extend this pattern match to include some more detailed info ?